addons/web_editor/static/src/js/editor/snippets.options.js

```diff
@@ -1024,7 +1024,11 @@ const ColorpickerUserValueWidget = SelectUserValueWidget.extend({
         if (typeof this._previewColor === 'string') {
             return this._previewColor;
         }
-        return this._super(...arguments);
+        let value = this._super(...arguments);
+        if (this.options.dataAttributes.hasOwnProperty('cssCompatible')) {
+            value = ColorpickerDialog.isCSSColor(value) ? value : `var(--${value})`;
+        }
```

Comment:
```suggestion
        if (this.options.dataAttributes.hasOwnProperty('cssCompatible')
                && !ColorpickerDialog.isCSSColor(value)) {
            value =  `var(--${value})`;
        }
```
Won't this be more logical to read this way?
And the condition smaller with `this.el.dataset.cssCompatible`?